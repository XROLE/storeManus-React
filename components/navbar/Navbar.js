/* eslint-disable class-methods-use-this */
import React, { Component } from 'react';
import { Redirect } from 'react-router';
import { Link } from 'react-router-dom';

class Navbar extends Component {
  constructor(props) {
    super();
  }

  logOut() {
    console.log('Login out');
    localStorage.removeItem('accessToken');
    window.location.replace('/');
  }

  render() {
    const token = localStorage.getItem('accessToken');
    return (
      <nav className="navbar">
        <div className="navLogo">
          <p>
            <Link to="/" style={{ color: 'white' }}>
              <i className="fas free fa-stream" />
              &nbsp;
              <span>Store Manus</span>
            </Link>
          </p>
          <p>
            <i className="fa fa-bars hanburgerButton" />
            <i className="fa fa-times close" />
          </p>
        </div>
        <ul className="nav-items">
          {token && <li onClick={e => this.logOut()} style={{ color: 'white' }} className="logOut">Logout</li>}
          { !token && <Link to="/login" style={{ color: 'white' }}><li>SignIn</li></Link> }
        </ul>
      </nav>
    );
  }
}

export default Navbar;
