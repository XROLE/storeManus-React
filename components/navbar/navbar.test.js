import React from 'react';
import { shallow } from 'enzyme';
import Navbar from './Navbar';

describe('Footer', () => {
  it('Should render just the Navbar component', () => {
    const component = shallow(
      <Navbar />,
    );
    expect(component.exists()).toBe(true);
    expect(component).toMatchSnapshot();
  });
});
