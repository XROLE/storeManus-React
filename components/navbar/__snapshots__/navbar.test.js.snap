// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Should render just the Navbar component 1`] = `
<nav
  className="navbar"
>
  <div
    className="navLogo"
  >
    <p>
      <Link
        replace={false}
        style={
          Object {
            "color": "white",
          }
        }
        to="/"
      >
        <i
          className="fas free fa-stream"
        />
        Â 
        <span>
          Store Manus
        </span>
      </Link>
    </p>
    <p>
      <i
        className="fa fa-bars hanburgerButton"
      />
      <i
        className="fa fa-times close"
      />
    </p>
  </div>
  <ul
    className="nav-items"
  >
    <Link
      replace={false}
      style={
        Object {
          "color": "white",
        }
      }
      to="/login"
    >
      <li>
        SignIn
      </li>
    </Link>
  </ul>
</nav>
`;
